src/Client.ts

```diff
@@ -263,7 +263,9 @@ export class Client {
             welcome = await this.connect(options.host, options.port)
         }
         if (useExplicitTLS) {
-            await this.useTLS(options.secureOptions)
+            let secureOptions = options.secureOptions || {}
+            secureOptions.host = secureOptions.host || options.host
```

Comment:
Okay, I'm willing to merge this if this really fixed things in your case. It did, right? There is one problem here we should be aware of: FTP uses two socket connections, one for commands (control) and one for each file transfer or directory listing (data). The error you're trying to fix here occurred with the data socket connection, not the control connection which is established initially, when using `access`. (I still don't know why this happens.) That `secureOptions` will be used later for setting up data socket connections. Now, the original FTP spec says that in theory an FTP server can make a client use an entirely different server for data connections than for the control one. With this change, we would probably no longer support that because we ourselves set the host... This is used very rarely today, though. Have to think about it but we could probably go ahead with this.

Reply:
To clarify, the error in the original issue is not showing up after the `useTLS` below. Only much later when an actual data transfer happens.